### PR TITLE
Enhance OrderState checking at install

### PR DESCRIPTION
**Pull request by Matt75**
Check if OrderState stored in Configuration exist and come from our module

### CODE REVIEW

File: src/OrderStates.php [hunk 1]
```diff
@@ -237,4 +237,22 @@ private function setStateIcons($state, $orderStateId)
...
+     *
+     * @param int $orderStateId
+     *
+     * @return bool
+     */
+    private function stateAlreadyExists($orderStateId)
+    {
+        $query = new \DbQuery();
+        $query->select('id_order_state');
+        $query->from(static::ORDER_STATE_TABLE);
```
Comment (L3RAZ) [inline src/OrderStates.php:252]: Maybe self:: should be used instead? Since it's used everywhere currently

File: src/OrderStates.php [hunk 2]
```diff
@@ -237,4 +237,22 @@ private function setStateIcons($state, $orderStateId)
...
+        $query->where('id_order_state = ' . (int) $orderStateId);
+        $query->where('module_name = "' . static::MODULE_NAME . '"');
```
Comment (L3RAZ) [inline src/OrderStates.php:254]: Same here for self::

File: src/OrderStates.php (hunk 1 above)
Comment (Matt75) [reply src/OrderStates.php]: done

File: src/OrderStates.php (hunk 2 above)
Comment (Matt75) [reply src/OrderStates.php]: done

**Review by L3RAZ — APPROVED** [review]

**Pull request merged**